synapse/logging/context.py

```diff
@@ -181,6 +180,42 @@ def __sub__(self, other: "ContextResourceUsage") -> "ContextResourceUsage":
         return res
 
 
+class ContextRequest:
```

Comment:
This could probably use `attrs`, but we don't seem to in other spots in this file...

Reply:
I think that's just because the file is old, can we make it an attrs please? That way we can have type hints